src/classes/RD2_RecurringDonations_TDTM.cls

```diff
@@ -155,11 +155,11 @@ public class RD2_RecurringDonations_TDTM extends TDTM_Runnable {
                 .revisePlannedInstallments();
 
             if (triggerAction == TDTM_Runnable.Action.BeforeInsert) {
-                rdRecord.reviseNextDonationDateBeforeInsert(scheduleService);
+                rdRecord
+                    .reviseNextDonationDateBeforeInsert(scheduleService);
 
             } else if (triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
                 rdRecord
-                    .reviseStatusForFixedLength()
```

Comment:
I can see why this method isn't called - because it's been merged into `reviseNextDonationDate()`, however in terms of "a method does one thing", would it be better if we kept that logic separate out so that it's explicit that the NDD calculation behavior doesn't also update Fixed Length RD status?

Reply:
I can split them, but then whenever the Next Donation Date is calculated, except on before `insert`, we need to call the fixed-length revise method. Thus, the change was done to minimize omission of the fixed-length check before the NDD calculations.